src/main/java/kdt/springbootboardjpa/service/PostService.java

```diff
@@ -0,0 +1,53 @@
+package kdt.springbootboardjpa.service;
+
+import jakarta.validation.Valid;
+import kdt.springbootboardjpa.respository.PostRepository;
+import kdt.springbootboardjpa.respository.UserRepository;
+import kdt.springbootboardjpa.respository.entity.Post;
+import kdt.springbootboardjpa.respository.entity.User;
+import kdt.springbootboardjpa.service.dto.PostDto;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+@Transactional(readOnly = true)
+public class PostService {
+
+    private final UserRepository userRepository;
+
+    private final PostRepository postRepository;
+
+    public List<Post> getAllPosts() {
+        return postRepository.findAll();
+    }
+
+    public Post getPostById(Long id) {
+        return postRepository.findById(id)
+                .orElseThrow(() -> new IllegalArgumentException("Can't find Post"));
+    }
+
+    @Transactional
+    public Post createPost(@Valid PostDto postDto) {
+        User user = userRepository.findById(postDto.createdBy())
+                .orElseThrow(() -> new IllegalArgumentException("Can't find User"));
+        Post newPost = Post.builder()
+                .title(postDto.title())
+                .content(postDto.content())
+                .user(user)
+                .build();
+        return postRepository.save(newPost);
+    }
+
+    @Transactional
+    public Post updatePost(Long postId, @Valid PostDto postDto) {
+        Post savedPost = postRepository.findById(postId)
+                .orElseThrow(() -> new IllegalArgumentException("Can't find Post"));
+        savedPost.updateTitle(postDto.title());
+        savedPost.updateContent(postDto.content());
```

Comment:
```suggestion
        savedPost.updateTitle(postDto.title());
        savedPost.updateContent(postDto.content());
```

두 필드에 대한 수정이 모두 일어난다면 update 메서드 하나만 만들고
 아예 엔티티 안에서 수정이 일어나도 좋을 것 같습니다.